episodes/11-parallel-intro.md

```diff
@@ -0,0 +1,101 @@
+complete their task totally on their own. They don't even have to be in the same room, they could be painting
+walls of different rooms in the house, in different houses in the city, and different cities in the
+country. We need, however, a communication system in place. Suppose that worker A, for some reason, needs a
+colour that is only available in the dispenser of worker B, they must then synchronise: worker A must request
+the paint of worker B and worker B must respond by sending the required colour.
+
+::::::::::::::::::::::::::::::::::::: callout
+
+Think of the memory distributed on each node of a cluster as the different dispensers for your workers
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+A **_fine-grained_** parallel code needs lots of communication or synchronisation between tasks, in contrast
+with a **_coarse-grained_** one. An **_embarrassingly parallel_** problem is one where all tasks can be
+executed completely independent from each other (no communications required).
+
+## Parallel programming in Chapel
+
+Chapel provides high-level abstractions for parallel programming no matter the grain size of your tasks,
+whether they run in a shared memory or a distributed memory environment, or whether they are executed
+concurrently or truly in parallel. As a programmer you can focus in the algorithm: how to divide the problem
+into tasks that make sense in the context of the problem, and be sure that the high-level implementation will
+run on any hardware configuration. Then you could consider the details of the specific system you are going to
+use (whether it is shared or distributed, the number of cores, etc.) and tune your code/algorithm to obtain a
+better performance.
+
+::::::::::::::::::::::::::::::::::::: callout
+
+To this effect, **_concurrency_** (the creation and execution of multiple tasks), and **_locality_** (in
+which set of resources these tasks are executed) are orthogonal concepts in Chapel.
+
+::::::::::::::::::::::::::::::::::::::::::::::::
+
+In summary, we can have a set of several tasks; these tasks could be running:
+
+1. concurrently by the same processor in a single compute node,
+2. in parallel by several processors in a single compute node,
+3. in parallel by several processors distributed in different compute nodes, or
+4. serially (one by one) by several processors distributed in different compute nodes.
+
+Similarly, each of these tasks could be using variables located in: 
+
+1. the local memory on the compute node where it is running, or 
+2. on distributed memory located in other compute nodes. 
+
+And again, Chapel could take care of all the stuff required to run our algorithm in most of the scenarios, but
+we can always add more specific detail to gain performance when targeting a particular scenario.
+
+::::::::::::::::::::::::::::::::::::: keypoints
+- "Concurrency and locality are orthogonal concepts in Chapel: where the tasks are running may not be
```

Comment:
Nothing about parallelizability?

Reply:
I added a key point about Chapel simplifying writing **_fine-grained_** parallel codes by hiding a lot of communication complexity under the hood. Not sure if this is what you had in mind mentioning parallelizability.